sycl/include/sycl/ext/oneapi/matrix/matrix-jit.hpp

```diff
@@ -451,6 +451,284 @@ class wi_element {
   }
 };
 
+// uint16_t here represents bf16 type as we have been doing for the
+// previous matrix implementations. Since the AMX and DPAS implementations don't
+// support uint16_t, this should raise no problem. Our plan is to move towards
+// SYCL bfloat16 once it makes itself to the specification (it is experimental
+// right now).
+template <size_t NumRows, size_t NumCols, matrix_layout Layout, typename Group>
+class wi_element<uint16_t, NumRows, NumCols, Layout, Group> {
+  joint_matrix<uint16_t, NumRows, NumCols, Layout, Group> &M;
+  std::size_t idx;
+
+public:
+  wi_element(joint_matrix<uint16_t, NumRows, NumCols, Layout, Group> &Mat,
+             std::size_t i)
+      : M(Mat), idx(i) {}
+  operator uint16_t() {
+#ifdef __SYCL_DEVICE_ONLY__
+    return __spirv_VectorExtractDynamic(M.spvm, idx);
+#else
+    throw runtime_error("joint matrix is not supported on host device.",
+                        PI_INVALID_DEVICE);
+#endif // __SYCL_DEVICE_ONLY__
+  }
+
+  explicit operator bool() {
+#ifdef __SYCL_DEVICE_ONLY__
+    return __spirv_VectorExtractDynamic(M.spvm, idx) !=
+           static_cast<uint16_t>(0);
+#else
+    throw runtime_error("joint matrix is not supported on host device.",
+                        PI_INVALID_DEVICE);
+#endif // __SYCL_DEVICE_ONLY__
+  }
+
+  wi_element &operator=(const uint16_t &rhs) {
+#ifdef __SYCL_DEVICE_ONLY__
+    M.spvm = __spirv_VectorInsertDynamic(M.spvm, rhs, idx);
+    return *this;
+#else
+    (void)rhs;
+    throw runtime_error("joint matrix is not supported on host device.",
+                        PI_INVALID_DEVICE);
+#endif // __SYCL_DEVICE_ONLY__
+  }
+
+  wi_element &
+  operator=(const wi_element<uint16_t, NumRows, NumCols, Layout, Group> &rhs) {
+#ifdef __SYCL_DEVICE_ONLY__
+    M.spvm = __spirv_VectorInsertDynamic(
+        M.spvm, __spirv_VectorExtractDynamic(rhs.M.spvm, rhs.idx), idx);
+    return *this;
+#else
+    (void)rhs;
+    throw runtime_error("joint matrix is not supported on host device.",
+                        PI_INVALID_DEVICE);
+#endif // __SYCL_DEVICE_ONLY__
+  }
+
+  // For now we use the following functions for convertion(bf16=>fp32,
+  // fp32=>bf16) as a workaround. In the future we will use
```

Comment:
Some rewording:
// We use here the following functions for conversion (bf16=>fp32 and fp32=>bf16). This is a workaround until we are able to use __spirv_ConvertFToBF16INTEL and __spirv_ConvertBF16ToFINTEL once these are supported in the CPU backend